### PR TITLE
Check if redisgraph is deployed before running tests

### PR DESCRIPTION
**Pull request by SherinV**
Deployment of RedisGraph will be disabled by default upstream.
This change enables the previous behavior before continuing with the tests.
Issue: https://github.com/open-cluster-management/backlog/issues/10931

### CODE REVIEW

File: tests/cypress/support/index.js
Comment (jlpadilla) [inline tests/cypress/support/index.js:45]: To enable the `log` task we need to add this to `/cypress/plugins/index.js`

```js
  on('task', {
    log(message) {
      console.log(message)

      return null
    },
  })
```

Comment (KevinFCormier) [reply tests/cypress/support/index.js:45]: Is there a reason not to use `cy.log`?

Comment (jlpadilla) [reply tests/cypress/support/index.js:45]: `cy.log()` doesn't print to the terminal so you can't see it when running headless in Travis.

File: tests/start-cypress-tests.sh
Comment (KevinFCormier) [inline tests/start-cypress-tests.sh:36]: This code will also run if search-redisgraph-0 is found but not running. Might be helpful to log the actual state here.

File: tests/cypress/support/index.js
Comment (KevinFCormier) [inline tests/cypress/support/index.js:48]: This version that runs inside Cypress does not include restarting search-collector. Is there a reason we have this both in the container entrypoint script and in Cypress, and is there a reason why the behaviour is slightly different in the two places?

**Review by KevinFCormier — COMMENTED** [review]
Added some questions and suggestions.

File: tests/start-cypress-tests.sh
Comment (SherinV) [reply tests/start-cypress-tests.sh:39]: Was intended as a TODO for search repo. We plan to move to search-operator later to deploy all pods, but currently use a combination of chart and operator.

File: tests/cypress/support/index.js
Comment (SherinV) [reply tests/cypress/support/index.js:48]: If the search-collector is running for a while without Redisgraph pod, it will go into a 10 minute time-out/wait/retry loop. We are only restarting the collector to avoid waiting for that long(10 minutes) to get a response. But, if we set the DEPLOY_REDISGRAPH env soon after the pods are setup, the Redisgraph pod will be available before the collector goes into longer 10 minute timeouts. So, we don't need to restart the collector in that case. Added in both cases to ensure that the pod is running, whichever way we trigger the test.

Comment (jlpadilla) [reply tests/cypress/support/index.js:48]: The reason for having both is because I'm not sure if app developers run cypress directly (without start-cypress-tests.sh). 
This one is more like a "safety net".  Here' we can't wait too long because it will cause the test to timeout.

File: tests/start-cypress-tests.sh
Comment (jlpadilla) [reply tests/start-cypress-tests.sh:35]: Yes, will change.

File: tests/start-cypress-tests.sh
Comment (jlpadilla) [reply tests/start-cypress-tests.sh:36]: @SherinV Now that we have the `deployredisgraph` status in our search operator, should use that instead of the pod status?

File: tests/start-cypress-tests.sh
Comment (jlpadilla) [reply tests/start-cypress-tests.sh:39]: Yes, I hope to complete this TODO before we merge. This is also the reason why the code inside Cypress doesn't do the restart because ideally the search operator will do it.

File: tests/start-cypress-tests.sh
Comment (jlpadilla) [inline tests/start-cypress-tests.sh:34]: Need to add `-n open-cluster-management` here.

File: tests/cypress/support/index.js
Comment (jlpadilla) [inline tests/cypress/support/index.js:43]: Need to add `-n open-cluster-management` here.

File: tests/start-cypress-tests.sh
Comment (SherinV) [reply tests/start-cypress-tests.sh]: Moved to operator in [PR](https://github.com/open-cluster-management/search-operator/pull/57)

**Review by KevinFCormier — APPROVED** [review]
Thanks for addressing all my concerns.

File: tests/start-cypress-tests.sh
Comment (SherinV) [reply tests/start-cypress-tests.sh]: Added. Thanks.

**Pull request merged**
